sql/src/test/java/io/crate/executor/transport/SnapshotRestoreDDLDispatcherTest.java

```diff
@@ -44,35 +48,57 @@
+        assertThat(ctx.resolvedTemplates, contains(".partitioned.my_table."));
     }
 
     @Test
     public void testResolveTableIndexFromSnapshot() throws Exception {
-        String resolvedIndex = SnapshotRestoreDDLDispatcher.ResolveFromSnapshotActionListener.resolveIndexNameFromSnapshot(
+        SnapshotRestoreDDLDispatcher.ResolveIndicesAndTemplatesContext ctx = new SnapshotRestoreDDLDispatcher.ResolveIndicesAndTemplatesContext();
+        SnapshotRestoreDDLDispatcher.ResolveFromSnapshotActionListener.resolveTableFromSnapshot(
             new TableIdent("custom", "restoreme"),
             Collections.singletonList(
                 new SnapshotInfo(new SnapshotId("snapshot01", UUID.randomUUID().toString()), Collections.singletonList("custom.restoreme"), 0L)
-            )
+            ),
+            ctx
         );
-        assertThat(resolvedIndex, is("custom.restoreme"));
+        assertThat(ctx.resolvedIndices, contains("custom.restoreme"));
+        assertThat(ctx.resolvedTemplates.size(), is(0));
     }
 
     @Test
     public void testResolvePartitionedTableIndexFromSnapshot() throws Exception {
-        String resolvedIndex = SnapshotRestoreDDLDispatcher.ResolveFromSnapshotActionListener.resolveIndexNameFromSnapshot(
+        SnapshotRestoreDDLDispatcher.ResolveIndicesAndTemplatesContext ctx = new SnapshotRestoreDDLDispatcher.ResolveIndicesAndTemplatesContext();
+        SnapshotRestoreDDLDispatcher.ResolveFromSnapshotActionListener.resolveTableFromSnapshot(
             new TableIdent(null, "restoreme"),
             Collections.singletonList(
                 new SnapshotInfo(new SnapshotId("snapshot01", UUID.randomUUID().toString()),
                     Collections.singletonList(".partitioned.restoreme.046jcchm6krj4e1g60o30c0"), 0L)
-            )
+            ),
+            ctx
         );
-        String template = PartitionName.templateName(null, "restoreme") + "*";
-        assertThat(resolvedIndex, is(template));
+        String template = PartitionName.templateName(null, "restoreme");
+        assertThat(ctx.resolvedIndices, contains(template + "*"));
+        assertThat(ctx.resolvedTemplates, contains(template));
+    }
+
+    @Test
+    public void testResolveEmptyPartitionedTemplate() throws Exception {
+        SnapshotRestoreDDLDispatcher.ResolveIndicesAndTemplatesContext ctx = new SnapshotRestoreDDLDispatcher.ResolveIndicesAndTemplatesContext();
+        SnapshotRestoreDDLDispatcher.ResolveFromSnapshotActionListener.resolveTableFromSnapshot(
+            new TableIdent(null, "restoreme"),
+            Collections.singletonList(
+                new SnapshotInfo(new SnapshotId("snapshot01", UUID.randomUUID().toString()), ImmutableList.of(), 0L)
+            ),
+            ctx
+        );
+        assertThat(ctx.resolvedIndices.size(), is(0));
+        assertThat(ctx.resolvedTemplates, contains(PartitionName.templateName(null, "restoreme")));
```

Comment:
pls add a comment why a template should be restored in that case, nothing beside the test name is partition related...